src/v1/sources/shopify/transform.js

```diff
@@ -0,0 +1,314 @@
+  if (event.updated_at) {
+    // converting shopify updated_at timestamp to rudder timestamp format
+    message.setTimestamp(new Date(event.updated_at).toISOString());
+  }
+  return message;
+};
+
+const ecomPayloadBuilder = (event, shopifyTopic) => {
+  const message = new Message(INTEGERATION);
+  message.setEventType(EventType.TRACK);
+  message.setEventName(RUDDER_ECOM_MAP[shopifyTopic]);
+
+  let properties = createPropertiesForEcomEvent(event);
+  properties = removeUndefinedAndNullValues(properties);
+  Object.keys(properties).forEach((key) =>
+    message.setProperty(`properties.${key}`, properties[key]),
+  );
+  // Map Customer details if present
+  const customerDetails = get(event, 'customer');
+  if (customerDetails) {
+    message.setPropertiesV2(customerDetails, MAPPING_CATEGORIES[EventType.IDENTIFY]);
+  }
+  if (event.updated_at) {
+    // TODO: look for created_at for checkout_create?
+    // converting shopify updated_at timestamp to rudder timestamp format
+    message.setTimestamp(new Date(event.updated_at).toISOString());
+  }
+  if (event.customer) {
+    message.setPropertiesV2(event.customer, MAPPING_CATEGORIES[EventType.IDENTIFY]);
+  }
+  if (event.shipping_address) {
+    message.setProperty('traits.shippingAddress', event.shipping_address);
+  }
+  if (event.billing_address) {
+    message.setProperty('traits.billingAddress', event.billing_address);
+  }
+  if (!message.userId && event.user_id) {
+    message.setProperty('userId', event.user_id);
+  }
+  return message;
+};
+
+const trackPayloadBuilder = (event, shopifyTopic) => {
+  const message = new Message(INTEGERATION);
+  message.setEventType(EventType.TRACK);
+  message.setEventName(SHOPIFY_TRACK_MAP[shopifyTopic]);
+
+  Object.keys(event)
+    .filter(
+      (key) =>
+        ![
+          'type',
+          'event',
+          'line_items',
+          'customer',
+          'shipping_address',
+          'billing_address',
+        ].includes(key),
+    )
+    .forEach((key) => {
+      message.setProperty(`properties.${key}`, event[key]);
+    });
+  // eslint-disable-next-line camelcase
+  const { line_items: lineItems, billing_address, user_id, shipping_address, customer } = event;
+  const productsList = getProductsListFromLineItems(lineItems);
+  message.setProperty('properties.products', productsList);
+  if (customer) {
+    message.setPropertiesV2(customer, MAPPING_CATEGORIES[EventType.IDENTIFY]);
+  }
+  // eslint-disable-next-line camelcase
+  if (shipping_address) {
+    message.setProperty('traits.shippingAddress', shipping_address);
+  }
+  // eslint-disable-next-line camelcase
+  if (billing_address) {
+    message.setProperty('traits.billingAddress', billing_address);
+  }
+  // eslint-disable-next-line camelcase
+  if (!message.userId && user_id) {
+    message.setProperty('userId', user_id);
+  }
+  return message;
+};
+
+const processEvent = async (inputEvent, metricMetadata, source) => {
```

Comment:
Dod we test identity stitching part here? How is it working? Considering we are not loading the tracker

Reply:
No, for now the main focus was on event transformation and delivery. We have some queued work to discuss and do on this going ahead

Reply:
ok